### PR TITLE
datafusion v28.0.0

### LINKED ISSUE #12: Readd macOS builds
In PR ( https://github.com/conda-forge/datafusion-feedstock/pull/10 ), macOS builds were disabled. Would be good to reenable them.

### PR DESCRIPTION
**Pull request by charlesbluca**
Attempting to debug the build errors we've been seeing in the past few release PRs

Closes #12
Closes #23
Closes #24
Closes #25
Closes #26

Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by charlesbluca** [comment]
@conda-forge-admin, please rerender

**Comment by jakirkham** [comment]
@conda-forge/datafusion, could you please review?

File: recipe/meta.yaml
Comment (charlesbluca) [inline recipe/meta.yaml]: cc @andygrove is there a reason we want to do this during the conda build versus the sdist/wheel builds? I would think we would want to include this in the files distributed on PyPI but could be missing something

Comment (andygrove) [reply recipe/meta.yaml]: It looks like you are referring to skipping Windows builds? There was an issue with Windows builds from PyPi that caused the conda builds to take too long so it was disabled, but this issue has now been resolved so we should no longer need to skip them.

Comment (charlesbluca) [reply recipe/meta.yaml]: I meant this command that I'm assuming is related to licensing:

```
cargo-bundle-licenses --format yaml --output THIRDPARTY.yml
```

It doesn't look like this is being done when we're building/distributing the source/wheels on PyPI, so I'm wondering if this is something we need specifically for conda-forge or if it would make sense to do this as part of the license generation job of the source/wheel builds:

https://github.com/apache/arrow-datafusion-python/blob/4175134cdf0ed5168282eb88feeb35a7540ac312/.github/workflows/build.yml#L27-L41

I've left things as is so this isn't a blocker for release, mostly wondering if this is a place where we could modify things moving forward

**Comment by rxm7706** [comment]
@conda-forge-admin, please rerender

**Comment by rxm7706** [comment]
Hi @andygrove, @cpcloud - I was looking to use ibis-datafusion and ended up here, could you review and merge this please. Thanks and Regards.

**Comment by rxm7706** [comment]
Hi @andygrove, @cpcloud - when you get a chance, could you review and merge this please. Thanks and Regards.

**Comment by charlesbluca** [comment]
@conda-forge-admin, please rerender

**Comment by andygrove** [comment]
@rxm7706 The PR has just been updated by @charlesbluca, and I plan on approving it later today. 

@cpcloud I asked Charles if he would mind becoming a maintainer here (we both work at NVIDIA) since I cannot effectively review PRs here. Let me know if you have any objections.

**Comment by charlesbluca** [comment]
@conda-forge-admin, please rerender

**Review by andygrove — APPROVED** [review]

**Comment by andygrove** [comment]
So, I don't have write access to this repo and cannot merge this PR. @cpcloud Could you review/merge when you have the time?

**Comment by andygrove** [comment]
I now have write access :tada:

**Pull request merged**
